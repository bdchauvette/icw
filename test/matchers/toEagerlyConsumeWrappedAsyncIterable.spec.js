test("passes when received value eagerly consumes wrapped async iterable", async () => {
  expect.assertions(1);

  expect(_ => eagerWrapper(_)).toEagerlyConsumeWrappedAsyncIterable();

  function eagerWrapper(iterable) {
    let iterator = iterable[Symbol.asyncIterator]();
    iterator.next();
  }
});

test("fails when received value does not consume wrapped async iterable", async () => {
  expect.assertions(2);

  await expect(
    expect(_ => noopWrapper(_)).toEagerlyConsumeWrappedAsyncIterable()
  ).rejects.toThrowErrorMatchingSnapshot();

  function noopWrapper() {}
});
