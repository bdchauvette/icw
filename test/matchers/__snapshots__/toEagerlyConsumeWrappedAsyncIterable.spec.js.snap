// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails when received value does not consume wrapped async iterable 1`] = `"expected wrapper to eagerly consume wrapped async iterable"`;
